content/docs/context.md

```diff
@@ -114,46 +112,89 @@ However, sometimes the same data needs to be accessible by many components in th
 ### `React.createContext`
 
 ```js
-const {Provider, Consumer} = React.createContext(defaultValue);
+const MyContext = React.createContext(defaultValue);
 ```
 
-Creates a `{ Provider, Consumer }` pair. When React renders a context `Consumer`, it will read the current context value from the closest matching `Provider` above it in the tree.
+Creates an Context object. When React renders a component that subscribes to this Context object it will read the current context value from the closest matching `Provider` above it in the tree.
 
-The `defaultValue` argument is **only** used by a Consumer when it does not have a matching Provider above it in the tree. This can be helpful for testing components in isolation without wrapping them. Note: passing `undefined` as a Provider value does not cause Consumers to use `defaultValue`.
+The `defaultValue` argument is **only** used when a component does not have a matching Provider above it in the tree. This can be helpful for testing components in isolation without wrapping them. Note: passing `undefined` as a Provider value does not cause consuming components to use `defaultValue`.
 
-### `Provider`
+### `Context.Provider`
 
 ```js
-<Provider value={/* some value */}>
+<MyContext.Provider value={/* some value */}>
 ```
 
-A React component that allows Consumers to subscribe to context changes.
+Every Context object comes with a Provider React component that allows consuming components to subscribe to context changes.
 
-Accepts a `value` prop to be passed to Consumers that are descendants of this Provider. One Provider can be connected to many Consumers. Providers can be nested to override values deeper within the tree.
+Accepts a `value` prop to be passed to consuming components that are descendants of this Provider. One Provider can be connected to many consumers. Providers can be nested to override values deeper within the tree.
 
-### `Consumer`
+All consumers that are descendants of a Provider will re-render whenever the Provider's `value` prop changes. The propagation from Provider to its descendant consumers is not subject to the `shouldComponentUpdate` method, so the consumer is updated even when an ancestor component bails out of the update.
```

Comment:
I wonder if we should clarify that context _is_ subject to a component's _own_ `shouldComponentUpdate` method, just not its ancestors. (Seems like this wording could be misleading.)